lib/options.ts

```diff
@@ -10,7 +10,7 @@ class Options {
   }
 
   public set(options: PkgOptions): void {
-    this.options = { ...this.options, ...options };
+    this.options = { ...this.options, ...(options ?? {}) };
```

Comment:
It's not very clear to me why you need to extend options if this is supposed to be called just once in the code?

Reply:
Because this can be undefined: `pkgOptions.set(inputJson?.pkg);`

So either I extend the option to make sure the default options made in constructor is not erased, or I do a condition like `this.options = options ?? this.options`

Reply:
I think it makes more sense to just write:

```ts
this.options = options ?? {}
```

Reply:
this is not valid as the dictionary key is mandatory, i didn't want to copy over the initial state multiple times, that's why i suggested `this.options = options ?? this.options`